vancouver-tree-height-geo.Rmd

```diff
@@ -62,55 +62,50 @@ str(trees) # maybe we don't use this
 
-Based on the data, it looks like there should be a one-to-one correspondence between levels of `HEIGHT_RANGE` and `HEIGHT_RANGE_ID`, but we should confirm this. To do this, we can look at all unique combinations of the two variables. If they properly correspond (there are no issues with the data), we should see only one row for each.
+Based on the data, it looks like there should be a one-to-one correspondence between levels of `HEIGHT_RANGE` and `HEIGHT_RANGE_ID`, but we should confirm this. To do this, we can look at all unique combinations of the two variables. If they properly correspond (there are no issues with the data), we should see only one row for each. We will print the results using `kable()` (from the `knitr` package).
 
 ```{r}
 unique_combinations <- trees |> 
   distinct(HEIGHT_RANGE, HEIGHT_RANGE_ID) |> 
   arrange(HEIGHT_RANGE_ID)
 
-unique_combinations |> kable()
-#TODO: add figure titles to this (in the kable code) and others
-
+unique_combinations |>
+  kable(caption = "Mapping between tree height category names and their IDs.")
 ```
 
-It looks like there is proper correspondence between levels of these two variables. Since the levels of `HEIGHT_RANGE_ID` are simpler and shorter, we will use this variable, which should make our plots easier to display.
+It looks like there is a proper correspondence between the levels of these two variables. Since the levels of `HEIGHT_RANGE_ID` are simpler and shorter, we will use this variable. This should make our plots easier to display.
 
-Now we can select only the treatment and response variable columns and convert them to type `factor`. Then, we can display the con
+Now we can select only the treatment and response variable columns and convert them to type `factor`, which will allow us to build a contingency table.
 
 ```{r}
 trees_subset <- trees |> 
   select(NEIGHBOURHOOD_NAME, HEIGHT_RANGE_ID) |> 
   mutate(across(everything(), as.factor))
 
-trees_subset
+head(trees_subset)
 ```
 
-Now, the `trees` tibble contains a row per tree, with only the columns `HEIGHT_RANGE` and `HEIGHT_RANGE_ID`.
-
-Generate a contingency table and print it using `kable()`
-
-NOTE TO SELF: add figure captions and for the one below, refer to the tibble above
+The new `trees_subset` dataframe contains one row per tree, with only the columns `NEIGHBOURHOOD_NAME` and `HEIGHT_RANGE_ID`. We can use this along with the `tabyl()` function from the `janitor` package to generate a contingency table. We will then print the contingency table using `kable()`.
 
 ```{r}
 cont_table <- trees_subset |> 
   tabyl(NEIGHBOURHOOD_NAME, HEIGHT_RANGE_ID)
 
 cont_table |> 
-  kable()
+  kable(caption = "Contingency table showing counts of trees in various levels of the tree height variable per levels of the neighbourhood variable. For a mapping of tree height category names, refer to Table 1.")
```

Comment:
maybe better to highlight where the Table 1 is

Reply:
The kable() function actually labels all the tables sequentually when we knit the PDF, so it all looks correct the way it is when we knit. Good catch, though.